### ISSUE TITLE
[mlir][pdll] Unknown command line argument '-no-warn-on-unused-template-args'

### ISSUE THREAD

**Issue by stellaraccident** [llvm/llvm-project#55716]
It seems that the `add_mlir_pdll_library` CMake macro delegates to the `tablegen` macro, which adds `-no-warn-on-unused-template-args` to invocations. I'm not sure how this is working upstream (is it?), but this breaks for me when including tests in a downstream. I was getting this just from running `ninja check-mlir` in a downstream.

I'm not sure it is the right fix, but I just added this to `mlir-pdll.cpp`'s main in my local branch:

```cpp
  // The tablegen main and CMake rules add this flag so we must accept it.
  llvm::cl::opt<bool> noWarnOnUnusedTemplateArg(
    "no-warn-on-unused-template-args",
    llvm::cl::desc("Disable unused template argument warnings."));
```

I expect this may be related to the FIXME on the previous line but have not looked closely (am working on something unrelated and just trying to unblock my build).
